compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1090,6 +1090,17 @@ object RefChecks {
 
   end checkImplicitNotFoundAnnotation
 
+  def checkSynchronizedCall(tree: Tree, symbol: Symbol)(using Context) =
+    if symbol.exists && (symbol.owner == defn.ObjectClass || symbol.owner == defn.AnyClass) then
```

Comment:
Perhaps we can add the last two stooges and go for
```suggestion
    if symbol.exists && defn.topClasses.contains(symbol.owner) then
```